cpp/src/parquet/arrow/reader.cc

```diff
@@ -219,6 +219,7 @@ class FileReaderImpl : public FileReader {
     ctx->iterator_factory = SomeRowGroupsFactory(row_groups);
     ctx->filter_leaves = true;
     ctx->included_leaves = included_leaves;
+    ctx->use_large_binary_variants = reader_properties_.use_large_binary_variants();
```

Comment:
Hmmmm, let me see if I understood the metadata thing correctly. While writing the parquet file, arrow will write something that indicates that `arrow::LargeBinary` type was used? And you want a test that validates that reading such file does not require the setting since it's already in the metadata?

Reply:
@arthurpassos Yes, that is what I mean in https://github.com/apache/arrow/pull/35825#pullrequestreview-1471347718

But this comment is just that, you've add logic in `GetColumnReader`, should same logic be applied in `GetReader`?

Reply:
I have fixed that already, thanks